Meta/CMake/sanitizers.cmake

```diff
@@ -1,24 +1,48 @@
+function(windows_clang_resource_lib_dir result_dir)
+    execute_process(
+            COMMAND ${CMAKE_CXX_COMPILER} -print-resource-dir
+            OUTPUT_VARIABLE CLANG_RESOURCE_DIR
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    file(TO_CMAKE_PATH "${CLANG_RESOURCE_DIR}" CLANG_RESOURCE_DIR)
+    set(${result_dir} "${CLANG_RESOURCE_DIR}/lib/windows" PARENT_SCOPE)
+endfunction()
+
 if (ENABLE_ADDRESS_SANITIZER)
-    add_cxx_compile_options(-fsanitize=address -fno-omit-frame-pointer)
-    add_cxx_link_options(-fsanitize=address)
-    add_swift_compile_options(-sanitize=address)
-    add_swift_link_options(-sanitize=address)
+    if (WIN32 AND CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
+        add_cxx_compile_options(-fsanitize=address /clang:-fno-omit-frame-pointer)
+        # FIXME: Windows on ARM
+        link_libraries(clang_rt.asan_dynamic-x86_64.lib -wholearchive:clang_rt.asan_dynamic_runtime_thunk-x86_64.lib)
+        windows_clang_resource_lib_dir(CLANG_WINDOWS_LIB_RESOURCE_DIR)
```

Comment:
`get_clang_resource_dir`  ? If you get just the base resource dir in the function, we could reuse that method on other platforms if necessary.